Core/src/Geometry/CuboidVolumeStack.cpp

```diff
@@ -0,0 +1,893 @@
+    ACTS_VERBOSE(*vt.updatedBounds);
+
+    vt.commit(logger);
+  }
+
+  ACTS_VERBOSE("*** Volume configuration after "
+               << axisDirectionName(m_dirOrth1) << "/"
+               << axisDirectionName(m_dirOrth2) << " synchronization:");
+  printVolumeSequence(volumeTuples, logger, Acts::Logging::VERBOSE);
+
+  std::ranges::sort(volumeTuples, {},
+                    [*this](const auto& v) { return v.mid(m_dir); });
+
+  m_volumes.clear();
+  for (const auto& vt : volumeTuples) {
+    m_volumes.push_back(vt.volume);
+  }
+
+  ACTS_DEBUG("*** Volume configuration after final " << axisDirectionName(m_dir)
+                                                     << " sorting:");
+  printVolumeSequence(volumeTuples, logger, Acts::Logging::DEBUG);
+
+  double min = volumeTuples.front().min(m_dir);
+  double max = volumeTuples.back().max(m_dir);
+
+  double mid = (min + max) / 2.0;
+  double hl = (max - min) / 2.0;
+
+  Translation3 translation(Vector3::Unit(dirIdx) * mid);
+  m_transform = m_groupTransform * translation;
+  std::array<double, 3> bounds;
+  bounds.at(dirIdx) = hl;
+  bounds.at(static_cast<std::size_t>(m_dirOrth1)) = hl1;
+  bounds.at(static_cast<std::size_t>(m_dirOrth2)) = hl2;
+
+  Volume::update(std::make_shared<CuboidVolumeBounds>(bounds), std::nullopt,
+                 logger);
+  ACTS_DEBUG("Outer bounds are:\n" << volumeBounds());
+  ACTS_DEBUG("Outer transform / new group transform is:\n"
+             << m_transform.matrix());
+
+  // Update group transform to the new center
+  // @TODO: We probably can reuse m_transform
+  m_groupTransform = m_transform;
+}
+
+void CuboidVolumeStack::overlapPrint(const CuboidVolumeStack::VolumeTuple& a,
+                                     const CuboidVolumeStack::VolumeTuple& b,
+                                     const Logger& logger) {
+  if (logger().doPrint(Acts::Logging::DEBUG)) {
+    std::stringstream ss;
+    ss << std::fixed;
+    ss << std::setprecision(3);
+    ss << std::setfill(' ');
+
+    int w = 9;
+
+    ACTS_VERBOSE("Checking overlap between");
+    ss << " - " << " " << axisDirectionName(m_dir) << ": [ " << std::setw(w)
+       << a.min(m_dir) << " <- " << std::setw(w) << a.mid(m_dir) << " -> "
+       << std::setw(w) << a.max(m_dir) << " ]";
+    ACTS_VERBOSE(ss.str());
+
+    ss.str("");
+    ss << " - " << " " << axisDirectionName(m_dir) << ": [ " << std::setw(w)
+       << b.min(m_dir) << " <- " << std::setw(w) << b.mid(m_dir) << " -> "
+       << std::setw(w) << b.max(m_dir) << " ]";
+    ACTS_VERBOSE(ss.str());
+  }
+}
+
+std::vector<CuboidVolumeStack::VolumeTuple>
+CuboidVolumeStack::checkOverlapAndAttach(
+    std::vector<VolumeTuple>& volumes,
+    CuboidVolumeStack::AttachmentStrategy strategy, const Logger& logger) {
+  // Preconditions: volumes are sorted along stacking direction
+  auto dirIdx = static_cast<std::size_t>(m_dir);
+  auto dirBoundIdx = static_cast<CuboidVolumeBounds::BoundValues>(m_dir);
```

Comment:
This should not be assumed. Access to these numbers should not rely on the internal integer value of these enums. The fact that they're the same should be considered *coincidence*.